File: 数据结构部分/1-100/_14.cpp
#include "normal.h"

/*
编写一个函数来查找字符串数组中的最长公共前缀。

如果不存在公共前缀，返回空字符串 ""。
*/

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.size() == 0)
            return "";

        string str = strs[0];
        int i, j;

        for (i = 1; i < strs.size(); i++) {
            for (j = 0; j < min(str.length(), strs[i].length()); j++) {
                if (str[j] != strs[i][j])
                    break;
            }

            if (j == 0)
                return "";

            if (j < str.length())
                str = str.substr(0, j);
        }

        return str;
     }
};

File: 数据结构部分/1-100/_14.py
"""
编写一个函数来查找字符串数组中的最长公共前缀。

如果不存在公共前缀，返回空字符串 ""。
"""

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        # Python中str比大小 
        # 比较两个字符串的从前到后的字符大小 按照字符的ASCII码比
        if not strs:
            return ""
        minStr = min(strs)
        maxStr = max(strs)        
        for i in range(len(minStr)):
            if minStr[i] != maxStr[i]:
                return minStr[:i]
        return minStr
